### ISSUE TITLE
Custom notification sound per contact

### ISSUE THREAD

**Issue by Enlargee** [FossifyOrg/Messages#241]
### Checklist

- [X] I made sure that there are **no existing issues** - [open](https://github.com/FossifyOrg/Messages/issues) or [closed](https://github.com/FossifyOrg/Messages/issues?q=is%3Aissue+is%3Aclosed) - to which I could contribute my information.
- [X] I made sure that there are **no existing discussions** - [open](https://github.com/FossifyOrg/Messages/discussions) or [closed](https://github.com/FossifyOrg/Messages/discussions?discussions_q=is%3Aclosed) - to which I could contribute my information.
- [X] I have read the FAQs inside the app (Menu -> About -> FAQs) and my problem isn't listed.
- [X] **I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.**
- [X] This issue contains only one feature request.
- [X] I have read and understood the [contribution guidelines](https://github.com/FossifyOrg/Messages/blob/master/CONTRIBUTING.md).
- [ ] I optionally [donated](https://fossify.org/donate) to support the Fossify mission.

### Feature description

Please allow the custom notification sounds to be assigned to a specific contact, vs being assinged to a conversation.

### Why do you want this feature?

Every 6-12 months, I clear out and archive my old SMS messages and store them on my PC. When I do that, I'll delete all my conversations as well and I think when that happens all the customised SMS tone assignments will then be deleted as well. Annoyingly, this isn't a feature in the stock contacts app any more. If this *doesn't* happen, and in fact they are saved - then please cancel/delete this request.

### Additional information

As it is, the last **must have** feature is present, and that's awesome to see. Thank you for creating such a well made app.

**Comment by seniordevel** [comment]
I would also like to see a custom notification sound, but I do not need this as 'per contact'.
Just a (custom) notification sound for messages (SMS) would be great.

**Comment by Aga-C** [comment]
@seniordevel It's already possible - check `Conversation Details`.

**Comment by seniordevel** [comment]
> It's already possible - check `Conversation Details`.

I could not find any setting. Just tried to re-check, but I deleted any older conversations.
Do you have a screenshot maybe?

**Comment by Aga-C** [comment]
@seniordevel You can find it in the three dots menu in the conversation for which you want to set notifications. If you mean to customize app-wide notifications, you can do it in Settings → Customize notifications.

**Comment by seniordevel** [comment]
> you can find it in the three dots menu in the conversation for which you want to set notifications. If you mean to customize app-wide notifications, you can do it in Settings → Customize notifications.

Again, I have any old conversations deleted!!
Thus the 3-dot menu just shows 'About' and 'Show archived conservatiens'. The latter is empty as expected.

Please read my answers carefully.

> you can do it in Settings → Customize notifications.
No! I posted this, because it is impossible on Android 11. There is no sound setting.

**Comment by antodc** [comment]
@seniordevel Next time you receive an SMS, go to the notification history, select messages and then the last notification. From there you should be able to change the system sound for SMSs.

**Comment by Enlargee** [comment]
Yes that's the ticket - I couldn't find this originally because I was starting with a blank list and it's not immediately obvious. You have to have at least one message from a contact, open that and the 3 dot menu has the relevant Conversation Details option.
 Then click Enable Custom Notifications
  -> then chose Customise Notifications
  -> then choose a sound from the options

This is prone to resetting your custom notification settings if you ever clear out your inbox of SMS, but that's something I do rarely enough that I'll manage with setting them again once I do that.

It would be extra nice if the app remembered custom settings per contact :)

**Comment by seniordevel** [comment]
> @seniordevel Next time you receive an SMS, go to the notification history, select messages and then the last notification. From there you should be able to change the system sound for SMSs.

Thank you.

But I think it probably will not work as expected, for the following reasons:
- As I understand, those sounds currently are assigned to conversations. But I am looking for an app-wide general notification sound, especially for completely new contacts, such as email notification sounds.
( @Aga-C commented: " you can do it in Settings → Customize notifications.", but there is no _sound_ notification setting)
- I usually delete any conversation after reading. As noted above, then the sound settings might be deleted too (as the  author guessed above)

However, lets see, I will try the feature and then report here.

**Comment by antodc** [comment]
> > you can do it in Settings → Customize notifications.
> 
> No! I posted this, because it is impossible, at least on on Android 11. There is no sound setting.

@seniordevel Just noticed. It is indeed true that there is no sound setting there. However, if you tap “SMS received” another notification setting view pops up. From the “advanced” section of that view you can change the notification sound.

**Comment by seniordevel** [comment]
> However, if you tap “SMS received” another notification setting view pops up. From the “advanced” section of that view you can change the notification sound.

Well, pretty good hidden Android setting, my great respect for finding it. Thank you!
Without having an opportunity to test it yet, I am pretty sure that this will work as I expect.

Maybe it would be a good idea to link this setting directly from the app, if possible.

**Comment by seniordevel** [comment]
> Anyway, I am curious to see your future report.

Today I can confirm this works fine. Now I have a customizable sound for any message incoming.

**Comment by mart3nsite** [comment]
Another similar request would be to respect the general notification settings for Sim 1 and Sim 2. For now, only one notification sound is used for both Sim and there is no app setting to change this behavior.
Maybe this request needs its own thread.

**Comment by Aga-C** [comment]
> Maybe this request needs its own thread.

@mart3nsite Yes, please create a new feature request for it.

**Comment by Enlargee** [comment]
I have found a way to have my SMS notification cake, and eat it too. 
Specifically, I can set a generic **SMS** sound that is NOT the **default system notification** sound.

To do this, I had to turn on a feature called "Manage notification categories for each app".
THEN
When you go to apps -> (Fossify) Messages -> Notifications -> (scroll down) -> Notification categories ->There'll be a long list (if you have several custom message sounds like I do, and one of them will be "Received SMS"

Set the received SMS category to your generic SMS sound. Set any other conversations to be whatever you want them to be, using the methods above (ie start a conversation, three dots, conversation details, notifications etc.)

**Comment by naveensingh** [comment]
I think adding a per-contact option will be a little advanced for this app. The current solution is sufficient for most use cases.

**Comment by Enlargee** [comment]
The current solution wasn't obvious back when the thread was created, and this reply I've just added, was here simply for those searching for how to do it.
